components/sidebar/sidebar_service.cc

```diff
@@ -72,6 +72,53 @@ SidebarItem::BuiltInItemType GetBuiltInItemTypeForLegacyURL(
   return SidebarItem::BuiltInItemType::kNone;
 }
 
+SidebarItem::BuiltInItemType GetPrevBuiltInItemFor(
+    const SidebarItem::BuiltInItemType& item) {
+  auto* iter_begin = std::cbegin(SidebarService::kDefaultBuiltInItemTypes);
+  auto* iter_end = std::cend(SidebarService::kDefaultBuiltInItemTypes);
+  auto* iter = std::find(iter_begin, iter_end, item);
+
+  if (iter_end == iter) {
+    return *std::prev(iter_end);
```

Comment:
`DCHECK` that the previous item is not `iter_begin`, i.e. there are items in `kDefaultBuiltInItemTypes`? Or just let this fall through to the logic below?

Reply:
How about adding this instead of this if clause?
```
CHECK(iter_end != iter);
```
With current constraint, `kDefaultBuiltInItemTypes` should include all.
```
      std::size(kDefaultBuiltInItemTypes) ==
          static_cast<size_t>(SidebarItem::BuiltInItemType::kBuiltInItemLast),
```